EasyVCR/CensorElement.cs

```diff
@@ -29,9 +30,69 @@ public CensorElement(string name, bool caseSensitive)
         /// </summary>
         /// <param name="key">The name to check.</param>
         /// <returns>True if the element matches, false otherwise.</returns>
-        internal bool Matches(string key)
+        internal virtual bool Matches(string key)
         {
-            return CaseSensitive ? Name.Equals(key) : Name.Equals(key, StringComparison.OrdinalIgnoreCase);
+            return CaseSensitive ? Value.Equals(key) : Value.Equals(key, StringComparison.OrdinalIgnoreCase);
+        }
+    }
+
+    public class RegexCensorElement : CensorElement
+    {
+        /// <summary>
+        ///     Constructor for a new regex censor element.
+        /// </summary>
+        /// <param name="pattern">Pattern of the element to censor.</param>
+        /// <param name="caseSensitive">Whether the pattern is case-sensitive.</param>
+        public RegexCensorElement(string pattern, bool caseSensitive) : base(pattern, caseSensitive)
+        {
+        }
+
+        /// <summary>
+        ///     Replace the provided value with the provided replacement if the value matches the regex pattern.
+        ///     Returns the original value if the value does not match the regex pattern.
+        /// </summary>
+        /// <param name="value">Value to apply the replacement to.</param>
+        /// <param name="replacement">Replacement for a detected matching section.</param>
+        /// <returns>The value with the replacement inserted, or the original value if no match was found.</returns>
+        internal string MatchAndReplaceAsNeeded(string value, string replacement)
+        {
+            var options = RegexOptions.Compiled | RegexOptions.CultureInvariant | RegexOptions.ExplicitCapture | RegexOptions.Singleline;
+            if (!CaseSensitive)
+            {
+                options |= RegexOptions.IgnoreCase;
+            }
+
+            return Regex.Replace(value,
+                Value,
+                replacement,
+                options,
+                TimeSpan.FromMilliseconds(250));
```

Comment:
Can we format like this instead? Having values on the same lines as the opening closing parentheses makes this confusing.

```suggestion
            return Regex.Replace(
                value,
                Value,
                replacement,
                options,
                TimeSpan.FromMilliseconds(250)
            );
```